### ISSUE TITLE
ISSN 1758-6623 | The ecumenical review (Wiley) | Verfasser

### ISSUE THREAD

**Issue by IxKo** [ubtue/DatenProbleme#1947]
#### URL
IxTheo#2023-04-08#98CDA3DD06888B49C4070B16B8814165237C3FFF
https://doi.org/10.1111/erev.12773
  
### Problembeschreibung

Ist es eventuell möglich solche Fälle über die QA-Settings mit einer passenden RegEx abzufangen?
Hier besteht der angebliche Nachname nur aus einem Buchstaben:

![image](https://user-images.githubusercontent.com/57000004/234209362-bd9a0146-131e-4277-8bb8-38400de43724.png)

**Comment by jriedl** [comment]
Man könnte für diese Zeitschrift für 100 und 700 eine erweiterte Form des Standard-Regex verwenden:

^(?!(?:(\s*(?:,\s*)?)|(\s*,\s*?\S+)|(\S(\.?)\s*,\s*\S+)|(.*,\s*[\p{P}-]\S+)<b>|(\s*\S\s*,.*)</b>)$)

hinterlegen.

**Comment by IxKo** [comment]
Da ein einzelner Buchstabe als Nachname auch bei anderen Zs. ein Indiz für fehlerhafte Einspielungen ist, sollte es global eingerichtet werden.

**Comment by jriedl** [comment]
Ja, aber die Abwägung sollte ja immer sein, für wieviele Fälle im Verhältnis zur Gesamtanzahl der innerhalb des Verfahrens generierten Records nehmen wir eine immer weiter zunehmende Komplexität und potentielle Seiteneffekte durch das Hinzufügen von Sonderregeln in Kauf.

**Comment by IxKo** [comment]
Würde so etwas nicht von den QA-Settings gefiltert werden, würden weitere solcher inkorrekten Daten zukünftig eingespielt werden.
Ich verstehe diese Debatte, um die Frage "Wie viele Fälle sind davon betroffen" daher nicht so recht.

Wenn es aber gewünscht ist, dass solche Fälle einfach eingespielt werden sollen, braucht es die QA-Settings gar nicht.

**Comment by jriedl** [comment]
> Es scheint eine Grundsatzfrage zu sein, ob man die Daten vorher abfängt oder sie teils eher zufällig im Nachhinein entdeckt und korrigiert. Die WinIBW ist in ihren Recherche-Möglichkeiten leider sehr eingeschränkt.
> 
Deshalb ja der Vorschlag, hier ein systematisches Verfahren zu entwickeln. Bei der Recherche auf unseren MARC-Daten sind wir ja nicht eingeschränkt und vielleicht genügt es ja schon, hier in regelmäßigen Abständen anhand des Abrufzeichens Text-Versionen der eingespielten Zotaut-Records zu erstellen, die man dann hoffentlich mit überschaubarem Aufwand durchsehen kann.
> Es gibt Vorgaben für die Belegung von den Verfasser:innen-Felder, weshalb die QA-Settings bei jeder Zs. sinnvoll wären. In meinen Augen wäre es daher gut, eine RegEx zu erstellen, die vorgibt, wie ein Name in die Marc-Felder 100 und 700 formal gestaltet werden darf. Vielleicht lag ich auch in meiner Annahme falsch, dass es leicht per RegEx umzusetzen wäre.

Da gab es seit Beginn des Verfahrens eine Unzahl von "endgültigen" Ansätzen, deshalb glaube ich da schlicht nicht mehr daran.

> Ich bin mir nicht sicher, ob dieser Fall sogar in den Errors gelandet wäre, wenn er nur einen Vornamen-Bestandteil gehabt hätte, bspw. "X, John" Eine globale RegEx existiert bereits für die Marc-Felder 100 und 700 und beim Testen schien sie "X, John" auszusortieren, nur nicht "X, Patriarch John" `^(?!(?:(\s*(?:,\s*)?)|(\s*,\s*?\S+)|(\S(\.?)\s*,\s*\S+)|(.*,\s*[\p{P}-]\S+))$)`

Ja, das war mir  auch aufgefallen, nur, wenn wir stattdessen
`/^(?!(?:(\s*(?:,\s*)?)|(\s*,\s*?\S+)|(\S(\.?)\s*,\s*\S+\s*\S+)|(.*,\s*[\p{P}-]\S+))$)`  verwenden, dann muss der Vorname zukünftig mindestens zwei Buchstaben haben, bei `^(?!(?:(\s*(?:,\s*)?)|(\s*,\s*?\S+)|(\S(\.?)\s*,\s*\S+\s+\S+)|(.*,\s*[\p{P}-]\S+))$)` wird zwar der aktuelle Fall erkannt, aber X, John wieder nicht mehr und was passiert bei noch mehr Namensbestandteilen? -  Wenn wir dafür `^(?!(?:(\s*(?:,\s*)?)|(\s*,\s*?\S+)|(\S(\.?)\s*,\s*[\S\s]+)|(.*,\s*[\p{P}-]\S+))$)` haben wir wieder ein Problem mit space-separierten Einzelbuchstaben usw. usf.

**Comment by jriedl** [comment]
Nachtrag: Um diese Diskussion nicht ausufern zu lassen und dem Wunsch einer situativen Anpassbarkeit Rechnung zu tragen, wäre eine alternative Option zu den obigen Anpassungsvorschlägen, den Default-Regex von `zotero_cgi` aus konfigurierbar zu machen.
